### PR TITLE
Quarkus Common Removal: Data API queryBuilder

### PR DESCRIPTION
**Pull request by Yuqi-Du**
Couple of notes
- Re-write builtCondition, eliminate the coupling with GRPC values
- Integrate the CQL string and CQL statement values into one building process, this will avoid we iterate through the filterExpression twice(reduce time complexity) and have a cleaner code
- Incorporate vector CQL query into new queryBuilder, this include the similarityScore, ANN order by and vector values
- added QueryBuilderTest unit tests, probably will add more complex tests.

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/cql/ExpressionUtils.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      List<? extends Expression<K>> expressions, String logicOperator) {
+    switch (logicOperator) {
+      case "$and" -> {
+        return andOf(expressions);
+      }
+      case "$or" -> {
+        return orOf(expressions);
+      }
+      default -> {
+        return null;
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/cql/ExpressionUtils.java:63]: Is this legal thing to happen? (if not, should throw exception)

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/CountOperation.java [hunk 2]
```diff
@@ -48,11 +47,11 @@ public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
   private SimpleStatement buildSelectQuery() {
     final List<Expression<BuiltCondition>> expressions =
         ExpressionBuilder.buildExpressions(logicalExpression, null);
-    List<Object> collect = new ArrayList<>();
-    if (expressions != null && !expressions.isEmpty() && expressions.get(0) != null) {
-      collect = ExpressionBuilder.getExpressionValuesInOrder(expressions.get(0));
-    }
-    QueryOuterClass.Query query = null;
+    //    List<Object> collect = new ArrayList<>();
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/CountOperation.java:50]: If these are meant to stay, need comment before to explain what it was doing; but probably should just be removed?

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java [hunk 3]
```diff
@@ -6,11 +6,12 @@
 import com.fasterxml.jackson.databind.node.ArrayNode;
 import com.fasterxml.jackson.databind.node.JsonNodeFactory;
 import com.fasterxml.jackson.databind.node.ObjectNode;
-import io.stargate.bridge.grpc.Values;
-import io.stargate.sgv2.api.common.cql.builder.BuiltCondition;
-import io.stargate.sgv2.api.common.cql.builder.Predicate;
+// import io.stargate.sgv2.api.common.cql.builder.BuiltCondition;
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java:9]: remove?

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java [hunk 4]
```diff
@@ -414,30 +412,22 @@ private List<SimpleStatement> buildSelectQueries(DBFilterBase.IDFilter additiona
     List<SimpleStatement> queries = new ArrayList<>(expressions.size());
     expressions.forEach(
         expression -> {
-          List<Object> collect = ExpressionBuilder.getExpressionValuesInOrder(expression);
+          //          List<Object> collect =
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java:415]: remove

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java [hunk 5]
```diff
@@ -531,17 +486,17 @@ private List<SimpleStatement> buildSortedSelectQueries(DBFilterBase.IDFilter add
...
               new QueryBuilder()
                   .select()
                   .column(columnsToAdd)
                   .from(commandContext.namespace(), commandContext.collection())
                   .where(expression)
                   .limit(maxSortReadLimit())
                   .build();
-          final SimpleStatement simpleStatement = SimpleStatement.newInstance(query.getCql());
-          queries.add(simpleStatement.setPositionalValues(collect));
+          queries.add(query.queryToStatement());
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java:499]: Nice! Looks like new way of building is much more compact from caller side. :)
(based on removal of codelines in this class)

File: src/test/java/io/stargate/sgv2/jsonapi/service/cql/builder/QueryBuilderTest.java [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+  @ParameterizedTest
+  @MethodSource("sampleQueries")
+  @DisplayName("Should generate expected CQL string")
+  public void generateCql(String actualCql, String expectedCql) {
+    assertThat(actualCql).isEqualTo(expectedCql);
+  }
+
+  @SuppressWarnings("PMD.ExcessiveMethodLength")
+  public static Arguments[] sampleQueries() {
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/service/cql/builder/QueryBuilderTest.java:35]: I may have missed this, but did not notice cases where quoting is needed for identifiers... when column name has mixed-case, for example? (quotes around identifier needed to preserve correct identifier casing)
Maybe add one or two cases?

**Review by tatu-at-datastax — APPROVED** [review]
LGTM: added some minor notes.

File: src/main/java/io/stargate/sgv2/jsonapi/service/cql/builder/Query.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package io.stargate.sgv2.jsonapi.service.cql.builder;
+
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import java.util.List;
+
+/**
+ * @param cql The query string. It can contain anonymous placeholders identified by a question mark
+ *     (?), or named placeholders prefixed by a column (:name).
+ * @param values The values to fill the placeholders in the query string.
+ * @param parameters The execution parameters for the query. (not supported)
```
[the remark this answers is not in the document]
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/cql/builder/Query.java:10]: thanks, deleted.

File: src/test/java/io/stargate/sgv2/jsonapi/service/cql/builder/QueryBuilderTest.java (hunk 6 above)
Comment (Yuqi-Du) [reply src/test/java/io/stargate/sgv2/jsonapi/service/cql/builder/QueryBuilderTest.java:39]: added!

**Pull request merged**